tests/containers/secret.pm

```diff
@@ -0,0 +1,89 @@
+# SUSE's openQA tests
+#
+# Copyright 2023-2024 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Package: podman, docker
+# Summary: Test the `secret` subcommand for Docker and Podman
+# Maintainer: QE-C team <qa-c@suse.de>
+
+use Mojo::Base 'containers::basetest';
+use testapi;
+use serial_terminal 'select_serial_terminal';
+use utils;
+use containers::common;
+use containers::container_images;
+
+sub run {
+    my ($self, $args) = @_;
+    my $runtime = $args->{runtime};
+    my $output = '';
+
+    select_serial_terminal();
+
+    # In Docker, secrets can only be used in a swarm
+    if ($runtime =~ 'docker') {
+        my $ip_addr = script_output(qq(ip -6 address show dev eth0 |
+        awk '/inet6/{split(\$2, a, "/"); print a[1]; exit;}'));
+        record_info("docker swarm init", "Initializing docker swarm in IP $ip_addr");
+        assert_script_run("docker swarm init --advertise-addr $ip_addr");
```

Comment:
I'd rather use `validate_script_output()` as this does not validate the output, only the return code.